pkg/apply/apply.go

```diff
@@ -194,6 +195,10 @@ func (t *TopicApplier) applyNewTopic(ctx context.Context) error {
 	return nil
 }
 
+func isFewerPartitionsError(err error) bool {
+	return strings.Contains(err.Error(), "Fewer partitions in topic config")
+}
```

Comment:
Should use `errors.Is`, add a sentinel error at the top like:

```go
var ErrFewerPartitions = errors.New("fewer paritions in topic config")
```

Then modify the `fmt.Errorf` to wrap this new sentinel error:

```go
return fmt.Errorf(
			"%w (%d) than observed (%d); this cannot be resolved by topicctl",
			ErrFewerPartitions,
			t.topicConfig.Spec.Partitions,
			currPartitions,
		)
```

Reply:
Sure, will use sentinel error :)

Reply:
btw, @jkoelker placing the error definition in the top of the file won't help:
The error should be raised from `pkg/apply/apply.go` while it should be checked in `cmd/topicctl/subcmd/apply.go`.
We need to place is a share pkg - I'd prefer a dedicated `errors` pkg.

Reply:
done.